File: clases 25-11-2025/Tipos de clases GO/clases_abstractas/Estudiante.go
// Estudiante.go
package main

import "fmt"

type Estudiante interface {
	Estudiar()
}

type estudiante struct {
	Nombre string
	Grado  int
}

type EstudianteReal struct {
	estudiante
}

func NewEstudianteReal(nombre string, grado int) Estudiante { // Mayúscula
	return EstudianteReal{estudiante{nombre, grado}}
}

func NewEstudianteRealFromOtro(otro Estudiante) Estudiante { // Mayúscula
	e := otro.(EstudianteReal)
	return NewEstudianteReal(e.Nombre, e.Grado)
}

func (e EstudianteReal) Estudiar() {
	fmt.Printf("%s está estudiando...\n", e.Nombre)
}

File: clases 25-11-2025/Tipos de clases GO/clases_abstractas/Estudiante.java
public abstract class Estudiante {
    public String nombre;
    public int grado;

    public Estudiante(String nombre, int grado) {
        this.nombre = nombre;
        this.grado = grado;
    }

    public Estudiante(Estudiante otro) {
        this.nombre = otro.nombre;
        this.grado = otro.grado;
    }

    public abstract void estudiar();
}